### PR TITLE
7072 create basic page

### PR DESCRIPTION
**Pull request by samMcCarthyBJSS**
Story [AB#7072](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7072)
Task [AB#8429](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8429)

image

### CODE REVIEW

File: app/helpers/api/bapi/getCatalogueItemPricing.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { getData } from 'buying-catalogue-library';
+import { getEndpoint } from '../../../endpoints';
+import { logger } from '../../../logger';
+
+export const getCatalogueItemPricing = async ({ accessToken, catalogueItemId, loggerText = 'Catalogue item' }) => {
+  const cataloguePricingEndpoint = getEndpoint({ api: 'bapi', endpointLocator: 'getCatalogueItemPricing', options: { catalogueItemId } });
```
Comment (alibombaye) [inline app/helpers/api/bapi/getCatalogueItemPricing.js:6]: as bit of a rule we are moving away from calling the getEndpoint anymore and moving that within here. see this as bit of an example

```
const getOrderItemEndpoint = (orderId, orderItemId) => (
  `${orderApiUrl}/api/v1/orders/${orderId}/order-items/${orderItemId}`
);

export const getOrderItem = async ({ orderId, orderItemId, accessToken }) => {
  const endpoint = getOrderItemEndpoint(orderId, orderItemId);
  const catalogueOrderItem = await getData({ endpoint, accessToken, logger });
  logger.info(`Catalogue order item returned for ${orderItemId}`);

  return catalogueOrderItem;
};
```

also remove `getCatalogueItemPricing` from the endpoints map

File: app/helpers/api/bapi/getCatalogueItemPricing.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { solutionsApiUrl } from '../../../config';
+import { logger } from '../../../logger';
+import {
+  getCatalogueItemPricing,
+} from './getCatalogueItemPricing';
+
+const accessToken = 'access_token';
+const catalogueItemId = 'associated-service-1';
+
+describe('findAssociatedServicePrices', () => {
```
Comment (alibombaye) [inline app/helpers/api/bapi/getCatalogueItemPricing.test.js:11]: change to `getCatalogueItemPricing`

File: app/pages/sections/order-items/additional-services/select/routes.test.js [hunk 3]
```diff
@@ -306,8 +308,7 @@ describe('additional-services select routes', () => {
     ));
 
     it('should call findAdditionalServicePrices once with the correct params if authorised', async () => {
```
Comment (alibombaye) [inline app/pages/sections/order-items/additional-services/select/routes.test.js:310]: call `getCatalogueItemPricing`

File: app/pages/sections/order-items/additional-services/select/price/controller.js [hunk 4]
```diff
@@ -1,24 +1,8 @@
...
-    accessToken,
-    logger,
-  });
-
-  logger.info(`Additional service pricing for additional service with ID: ${catalogueItemId} found in BAPI.`);
-
-  return additionalServicePricingData;
-};
-
 export const validateAdditionalServicePriceForm = ({ data }) => {
```
Comment (alibombaye) [inline app/pages/sections/order-items/additional-services/select/price/controller.js:6]: Nice.. is there one in catalogue-solutions/select/price that can also be removed

Comment (samMcCarthyBJSS) [reply app/pages/sections/order-items/additional-services/select/price/controller.js:6]: nah that uses getSolutionPricing

File: app/helpers/api/bapi/getCatalogueItemPricing.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { getData } from 'buying-catalogue-library';
+import { logger } from '../../../logger';
+import { solutionsApiUrl } from '../../../config';
+
+export const getCatalogueItemPricingEndpoint = ({ catalogueItemId }) => (
+  `${solutionsApiUrl}/api/v1/prices?catalogueItemId=${catalogueItemId}`
+);
```
Comment (alibombaye) [inline app/helpers/api/bapi/getCatalogueItemPricing.js:7]: 👍

File: app/pages/sections/order-items/additional-services/select/price/controller.js (hunk 4 above)
Comment (alibombaye) [reply app/pages/sections/order-items/additional-services/select/price/controller.js:6]: ok I think that can be changed to use this too. But leave as is for now. I'll take a look later 👍

**Pull request merged**